udp_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2020 Storj Labs, Inc.
+// Copyright (C) 2014 Space Monkey, Inc.
+// See LICENSE for copying information.
+
+package jaeger
+
+import (
+	"testing"
+	"time"
+
+	"github.com/spacemonkeygo/monkit/v3"
+	"github.com/stretchr/testify/require"
+
+	"storj.io/monkit-jaeger/gen-go/jaeger"
+)
+
+func TestUDPCollector(t *testing.T) {
+	agentAdrr := "127.0.0.1:0"
+	mock, err := StartMockAgent(agentAdrr)
+	require.NoError(t, err)
+	defer func() {
+		err := mock.Close()
+		require.NoError(t, err)
+	}()
+	collector, err := NewUDPCollector(mock.Addr(), 200, "test", nil)
+	require.NoError(t, err)
+
+	span := &jaeger.Span{
+		TraceIdLow:    monkit.NewId(),
+		SpanId:        monkit.NewId(),
+		OperationName: "test-udp-collector",
+		StartTime:     time.Now().UnixNano() / 1000,
+		Duration:      time.Second.Microseconds(),
+	}
+	collector.Collect(span)
+
+	data := mock.GetBatches()
+	for len(data) < 1 {
+		time.Sleep(1 * time.Millisecond)
```

Comment:
Why is this sleep here? Sleeps are notorious for causing flaky tests. In 99.99% of cases, the right answer is to redesign the test so it does not need a sleep.

Reply:
I see better now, you're waiting for it to occur. In that case, you at least need a max time, so that you don't have an infinite loop in the test.